internal/datacoord/compaction_trigger.go

```diff
@@ -249,7 +294,35 @@ func (t *compactionTrigger) triggerCompaction() error {
 	return nil
 }
 
-// triggerSingleCompaction triger a compaction bundled with collection-partition-channel-segment
+// triggerClusteringCompaction trigger clustering compaction.
+func (t *compactionTrigger) triggerClusteringCompaction() error {
```

Comment:
I think we need to make sure mutual exclusion between clustering compaction and L0/mix compaction. You can integrate this logic in compaction_scheduler.go.